### ISSUE TITLE
SIGSEGV created with api accessing streaming info

### ISSUE THREAD

**Issue by slivne** [scylladb/scylla#788]
using head 

scylla

e88f41fb3f64e7ffc514b99ebab091c6999a98cd

jmx

d0901360853c07ee9eccf4826a02decc01700914

```
-sh-4.3$ sudo coredumpctl info 33705
           PID: 33705 (scylla)
           UID: 800 (jenkins)
           GID: 1012 (jenkins)
        Signal: 11 (SEGV)
     Timestamp: Wed 2016-01-13 11:11:47 IST (1h 25min ago)
  Command Line: /data2/jenkins/.dtest/dtest-_E_rXF/test/node2/bin/scylla --options-file /data2/jenkins/.dtest/dtest-_E_rXF/test/node2/conf/scylla.yaml --api-address 127.0.0.2 --collectd-hostname monster.node2 --
    Executable: /data2/jenkins/.dtest/dtest-_E_rXF/test/node2/bin/scylla
 Control Group: /user.slice/user-800.slice/session-927.scope
          Unit: session-927.scope
         Slice: user-800.slice
       Session: 927
     Owner UID: 800 (jenkins)
       Boot ID: 4b4b25aed1d34ffbbe18440f78bce3f2
    Machine ID: 2e99f913871b4635bcca4758b6de2f2d
      Hostname: monster
      Coredump: /var/lib/systemd/coredump/core.scylla.800.4b4b25aed1d34ffbbe18440f78bce3f2.33705.1452676307000000
       Message: Process 33705 (scylla) of user 800 dumped core.
                
                Stack trace of thread 33705:
                #0  0x000000000112bbbc _ZN10shared_ptrIN9streaming14stream_managerEEC4ERKS2_ (scylla)
                #1  0x0000000001126926 operator() (scylla)
                #2  0x00000000011276c8 submit_to<seastar::sharded<Service>::map_reduce0(Mapper, Initial, Reduce)::<lambda(unsigned int)> [with Mapper = api::set_stream_manager(api::http_context&, httpd::routes&)
                #3  0x000000000059f125 _ZNKSt8functionIF6futureIJN4json16json_return_typeEEESt10unique_ptrIN5httpd7requestESt14default_deleteIS6_EEEEclES9_ (scylla)
                #4  0x00000000005a11cd _ZNKSt8functionIF6futureIJSt10unique_ptrIN5httpd5replyESt14default_deleteIS3_EEEES1_INS2_7requestES4_IS8_EES6_EEclESA_S6_ (scylla)
                #5  0x00000000005aeee6 _ZN5httpd6routes6handleERK13basic_sstringIcjLj15EESt10unique_ptrINS_7requestESt14default_deleteIS6_EES5_INS_5replyES7_ISA_EE (scylla)
                #6  0x000000000061f88f _ZN5httpd11http_server10connection14generate_replyESt10unique_ptrINS_7requestESt14default_deleteIS3_EE (scylla)
                #7  0x000000000061fd96 _ZZZN5httpd11http_server10connection8read_oneEvENUlvE_clEvENUlvE_clEv (scylla)
                #8  0x000000000061ff57 _ZN6futureIJEE4thenIZZN5httpd11http_server10connection8read_oneEvENUlvE_clEvEUlvE_S_IJbEEEET0_OT_ (scylla)
                #9  0x00000000006202a9 _ZZN5httpd11http_server10connection8read_oneEvENUlvE_clEv (scylla)
                #10 0x000000000062045c _ZN12apply_helperIZN5httpd11http_server10connection8read_oneEvEUlvE_OSt5tupleIJEESt16integer_sequenceImJEEE5applyEOS3_S6_ (scylla)
                #11 0x0000000000470b9d _ZN7reactor9run_tasksER15circular_bufferISt10unique_ptrI4taskSt14default_deleteIS2_EESaIS5_EE (scylla)
                #12 0x000000000049b17b _ZN7reactor3runEv (scylla)
                #13 0x000000000050c99a _ZN12app_template14run_deprecatedEiPPcOSt8functionIFvvEE (scylla)
                #14 0x000000000041711e main (scylla)
                #15 0x00007f1af007f580 __libc_start_main (libc.so.6)
                #16 0x000000000046d119 _start (scylla)
                
                Stack trace of thread 33708:
                #0  0x00007f1af0159947 ftruncate64 (libc.so.6)
                #1  0x000000000046e859 operator() (scylla)
                #2  0x00000000004714df operator() (scylla)
                #3  0x00000000004f86ce _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f1af042760a start_thread (libpthread.so.0)
                #5  0x00007f1af0161a9d __clone (libc.so.6)
```

**Comment by penberg** [comment]
There's a (hopefully) temporary workaround for this issue: https://github.com/scylladb/scylla/commit/733584c44df07f231021944cfdda1e88ee1dd6f0

**Comment by slivne** [comment]
fixed by b1845cddeceaa5a84509b1827243bbf472b669ba
